src/SamplesApp/SamplesApp.iOS/Class1.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Foundation;
+using Microsoft.Extensions.DependencyInjection;
+using ObjCRuntime;
+
+namespace Microsoft.Extensions.Logging.NSLogger
+{
+	/// <summary>
+	/// A logger that writes messages in the debug output window only when a debugger is attached.
+	/// </summary>
+	public partial class NSDebugLogger : ILogger
+	{
+		private readonly Func<string, LogLevel, bool> _filter;
+		private readonly string _name;
+
+		[DllImport(Constants.FoundationLibrary)]
+		extern static void NSLog(IntPtr format, IntPtr s);
+
+		static NSString format = new NSString("%@");
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="DebugLogger"/> class.
+		/// </summary>
+		/// <param name="name">The name of the logger.</param>
+		public NSDebugLogger(string name) : this(name, filter: null)
+		{
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="DebugLogger"/> class.
+		/// </summary>
+		/// <param name="name">The name of the logger.</param>
+		/// <param name="filter">The function used to filter events based on the log level.</param>
+		public NSDebugLogger(string name, Func<string, LogLevel, bool> filter)
+		{
+			_name = string.IsNullOrEmpty(name) ? nameof(NSDebugLogger) : name;
+			_filter = filter;
+		}
+
+
+		/// <inheritdoc />
+		public IDisposable BeginScope<TState>(TState state)
+		{
+			return NoopDisposable.Instance;
+		}
+
+		/// <inheritdoc />
+		public bool IsEnabled(LogLevel logLevel)
+		{
+			// If the filter is null, everything is enabled
+			// unless the debugger is not attached
+			return Debugger.IsAttached &&
+				logLevel != LogLevel.None &&
+				(_filter == null || _filter(_name, logLevel));
+		}
+
+		/// <inheritdoc />
+		public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+		{
+			if (!IsEnabled(logLevel))
+			{
+				return;
+			}
+
+			if (formatter == null)
+			{
+				throw new ArgumentNullException(nameof(formatter));
+			}
+
+			var message = formatter(state, exception);
+
+			if (string.IsNullOrEmpty(message))
+			{
+				return;
+			}
+
+			message = $"{ logLevel }: {message}";
+
+			if (exception != null)
+			{
+				message += Environment.NewLine + Environment.NewLine + exception.ToString();
```

Comment:
Codacy found an issue: [There's no need to call 'ToString()', the compiler will do it for you.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5982976)